src/layer/tile/GridLayer.js

```diff
@@ -444,8 +444,9 @@ L.GridLayer = L.Layer.extend({
 
 	_getTiledPixelBounds: function (center) {
 		var map = this._map,
+		    scale = map.getZoomScale(map.getZoom(), this._tileZoom),
 		    pixelCenter = map.project(center, this._tileZoom).floor(),
-		    halfSize = map.getSize().divideBy(2);
+		    halfSize = map.getSize().divideBy(scale * 2);
```

Comment:
Scaling the size does not make sense when you think about it: the map *size* does not change from one zoom to another. :)

So what this revert do is make Leaflet request too much tiles (like 4 times too much in some round cases), so it seems to be fixing the issue but it's not ;)